source/manual/alerts/rabbitmq-no-consumers-listening.html.md

```diff
@@ -4,75 +4,94 @@ title: 'RabbitMQ: No consumers listening to queue'
+to check on the activity of the consumers of a given RabbitMQ message queue.
 
-If the check fails due to there being no consumers connected to the queue, it
-will report a critical failure:
+Publishing API [sends a heartbeat message][heartbeat_messages] every minute to
+the following queues - `email-alert-service`, `cache_clearing_service-high` and
+`content_data_api`. These queues receive the heartbeat due to being classed as
+particularly important. This is configured via the queues bindings (e.g
+[email-alert-service's binding][email_alert_binding]) matching the routing key
+used in the [heartbeat][heartbeat]. This should ensure that the consumers are
+never idle for these queues.
 
-```
-CRITICAL: "No consumers listening to queue"
-```
+## Troubleshooting
+
+### Check the RabbitMQ logs
+
+Check the logs for the `rabbitmq` application which can be achieved logging
+into Kibana and searching for `application: rabbitmq`. Is there evidence of any
+errors?
+
+### Check the RabbitmQ Grafana dashboard
+
+This [Grafana dashboard][rabbitmq_grafana_dashboard] shows activity across
+multiple exchanges and queues. The main exchange we expect to be monitoring is
+`published_documents` which handles broadcasting to services such as search and
+email-alert-service when content changes across GOV.UK.
+
+Looking at the queue graphs we should look out for the following:
+
+* **Check for high 'ready' messages for the queue** - indicates these messages are
+  waiting to be processed in RabbitMQ by the consumer (e.g
+  email-alert-service).
 
-## Consequences of idle consumers
+* **Check for high 'unacknowledged' messages for the queue** - implies that
+  messages have been read by the consumer but the consumer has never sent back
+  an ACK to the RabbitMQ broker to say that it has finished processing.
 
-If consumers are idle, messages sent to the queue will not be being processed.
-This means that the actions which are meant to happen in response to the
-messages will not happen. For example, email alerts about updated content will
-not be sent if the `email_alert_sevice` queue isn't running.
+* **Check for high 'redeliver' rate for the queue** - in the event of network
+  failure (or a node failure), messages can be redelivered. An example is if
+  the consumer dies (its channel is closed, connection is closed, or TCP
+  connection is lost).
 
-Unless there is a wider RabbitMQ failure, messages will not be lost - they will
-be processed once the problem is resolved.
+If we're seeing high 'redeliver' rates, high 'ready' or 'unacknowledged'
+messages then this could indicate an issue with the consumer.
 
-## Investigation
+### Troubleshooting steps
 
-RabbitMQ has an admin interface, which allows details of recent activity on the
-queues to be seen. To log in, [follow the instructions in the RabbitMQ
-docs][rabbitmq_control_panel].
+1. Reboot all RabbitMQ nodes, you can do this safetly by following the steps
```

Comment:
Noooooo! This would probably make it worse. I was actually referring to restarting the consumers processes, which can sometimes get stuck _when RabbitMQ reboots_.

Reply:
Mentioning restarting and linking to a rebooting guide made me assume you meant rebooting RabbitMQ. 🙈 

Oh well, I've added what you were actually thinking. 😄

Reply:
Ah, sorry about that. The bit I was referring to was:

> Publishing API jobs became stuck - This required restarting Publishing API workers to resolve.

But I see now how it's confusing, as the following looks more relevant:

> No non-idle RabbitMQ consumers - This required killing RabbitMQ processes to resolve.

Looking in the incident report, I think this was something of a one-off due to a planned reboot going wrong. While the alert was triggered, the actual problem was with RabbitMQ itself, and this alert is only tangentially related.